### ISSUE TITLE
use gherkin3

### ISSUE THREAD

**Issue by ahdinosaur** [yoshuawuyts/gherkin-parser#1]
`gherk` only supports a subset of Gherkin syntax, meanwhile [`gherkin`](https://www.npmjs.com/package/gherkin-javascript) is a fancy new fully-featured implementation as part of [gherkin3](https://github.com/cucumber/gherkin3). i think it'd be sweet if we used gherkin3 here.

**Comment by yoshuawuyts** [comment]
Yeah, definitely. :sparkles: Adding you as a contrib to both repo + npm. PRs very much welcome!

**Comment by ahdinosaur** [comment]
hmm, looking into it it seems we can even do proper streaming if we can [provide `gherkin.Parser#parse`](https://github.com/cucumber/gherkin3/blob/master/javascript/lib/gherkin/parser.js#L51-L54) with an interface compatible with [`gherkin.TokenScanner`](https://github.com/cucumber/gherkin3/blob/master/javascript/lib/gherkin/token_scanner.js), which conveniently means [implementing a `read()` method that returns each line](https://github.com/mcollina/split2).

**Comment by yoshuawuyts** [comment]
woah, that's heaps good

**Comment by ahdinosaur** [comment]
ah, turns out we can't do this asynchronously as a stream because each `read()` call is synchronous. but i reckon that's fine, at most we're buffering a feature file which isn't much.
